lib/src/Pos/Network/Block/Retrieval.hs

```diff
@@ -114,7 +114,7 @@ retrievalWorker pm txpConfig diffusion = do
         logDebug $ "handleContinues: " <> pretty hHash
         classifyNewHeader pm header >>= \case
             CHContinues ->
-                void $ getProcessBlocks pm txpConfig diffusion nodeId (headerHash header) [hHash]
+                void $ getProcessBlocks pm txpConfig diffusion nodeId hHash [hHash]
```

Comment:
What problem is this fixing? To me this actually looks incorrect. The function:
```
getProcessBlocks
    :: forall ctx m.
       ( BlockWorkMode ctx m
       , HasMisbehaviorMetrics ctx
       )
    => ProtocolMagic
    -> TxpConfiguration
    -> Diffusion m
    -> NodeId
    -> HeaderHash
    -> [HeaderHash]
    -> m ()
getProcessBlocks pm txpConfig diffusion nodeId desired checkpoints = do
```
which suggests that the second last parameter is *different* from the last parameter.

Reply:
Look at line 113, the hash is already calculated, it's intent to download a single block, so these parameters are the same.